projects/igniteui-angular/src/lib/drop-down/drop-down-utils.ts

```diff
@@ -0,0 +1,72 @@
+import { IToggleView } from '../core/navigation/IToggleView';
+import { OnInit, EventEmitter, ElementRef } from '@angular/core';
+import { ISelectionEventArgs, Navigate } from './drop-down.common';
+import { CancelableEventArgs } from '../core/utils';
+import { IgxComboBase } from '../combo/combo.common';
+import { DropDownActionKeys } from './drop-down-navigation.directive';
+
+export interface DropDownNavigationDirective {
+    target: any;
+    handleKeyDown(event: KeyboardEvent): void;
+    onArrowDownKeyDown(event?: KeyboardEvent): void;
+    onArrowUpKeyDown(event?: KeyboardEvent): void;
+    onEndKeyDown(event?: KeyboardEvent): void;
+    onHomeKeyDown(event?: KeyboardEvent): void;
+}
+
+export interface IDropDownList extends IToggleView, OnInit {
+    onOpening: EventEmitter<CancelableEventArgs>;
+    onOpened: EventEmitter<void>;
+    onClosing: EventEmitter<CancelableEventArgs>;
+    onClosed: EventEmitter<void>;
+    width: string;
+    height: string;
+    id: string;
+    cssClass: string;
+    maxHeight: string;
+    collapsed: boolean;
+    items: IDropDownItem[];
+    focusedItem: IDropDownItem;
+    navigateFirst(): void;
+    navigateLast(): void;
+    navigateNext(): void;
+    navigatePrev(): void;
+    onToggleOpening(e: CancelableEventArgs): void;
+    onToggleOpened(): void;
+    onToggleClosing(e: CancelableEventArgs): void;
+    onToggleClosed(): void;
+    navigateItem(newIndex: number, direction?: Navigate): void;
+    handleKeyDown(key: DropDownActionKeys);
+}
+
+
+export const IGX_DROPDOWN_BASE = 'IgxDropDownBaseToken';
+export interface IDropDownBase extends IDropDownList {
+    onSelection: EventEmitter<ISelectionEventArgs>;
+    combo?: IgxComboBase;
+    comboID?: string;
```

Comment:
Same reasoning as the combo - this shoudln't be here at all. I see items rely on it to use the service, which should just be per-instance and drop the component map entirely.